src/HeaderBar.vala

```diff
@@ -47,6 +50,37 @@ public class Mail.HeaderBar : Hdy.HeaderBar {
             valign = Gtk.Align.CENTER
         };
 
+        var hide_read_button = new Granite.SwitchModelButton (_("Unread")) {
+            description = _("Hides all read emails")
+        };
+
+        var hide_unstarred_button = new Granite.SwitchModelButton (_("Starred")) {
+            description = _("Hides all unstarred emails")
+        };
+
+        var filter_menu_popover_grid = new Gtk.Grid () {
+            column_spacing = 6,
+            margin_bottom = 6,
+            margin_top = 12,
+            orientation = Gtk.Orientation.VERTICAL,
+            row_spacing = 6
+        };
+
+        filter_menu_popover_grid.add (hide_read_button);
+        filter_menu_popover_grid.add (hide_unstarred_button);
+        filter_menu_popover_grid.show_all ();
+
+        var filter_popover = new Gtk.Popover (null);
+        filter_popover.add (filter_menu_popover_grid);
+
+        filter_button = new Gtk.MenuButton () {
+            can_focus = false,
```

Comment:
We don't want to prevent buttons from being able to gain focus. This would make it inaccessible with the keyboard
```suggestion
```